app/controllers/v0/map_services_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'map/security_token/service'
+
+module V0
+  class MapServicesController < SignIn::ApplicationController
+    skip_before_action :authenticate
+    before_action :authenticate_service_account
+
+    # POST /v0/map_services/:application/token
+    def token
+      unless (icn = @service_account_access_token.user_attributes['icn'])
+        render json: missing_icn_error, status: :bad_request and return
+      end
+
+      result = MAP::SecurityToken::Service.new.token(application: params[:application].to_sym, icn:)
+
+      render json: result, status: :ok
+    rescue Common::Client::Errors::ClientError
+      render json: sts_client_error, status: :bad_gateway
+    rescue MAP::SecurityToken::Errors::ApplicationMismatchError
+      render json: application_mismatch_error, status: :bad_request
+    end
+
+    private
+
+    def sts_client_error
+      {
+        error: 'server_error',
+        error_description: 'STS failed to return a valid token.'
+      }
+    end
+
+    def application_mismatch_error
+      {
+        error: 'invalid_request',
+        error_description: "Application mismatch detected. Application: #{params[:application]}"
```

Comment:
We don't need to return the application the caller just passed in, `Application Mismatch detected` should be sufficient